File: packages/web/libs/api/calendar.ts
import { request } from '@/libs/request'

const api = request()

export interface CalendarEvent {
  id: string
  title: string
  start: string
  end: string
  all_day: boolean
  type: 'meeting' | 'task'
  status?: string
  metadata: {
    original_id: string
    [key: string]: any
  }
}

export async function getCalendarEvents(start: string, end: string): Promise<CalendarEvent[]> {
  return api.get<CalendarEvent[]>('/v1/calendar/events', {
    params: { start_date: start, end_date: end },
  })
}

File: packages/api/app/schemas/calendar.py
from __future__ import annotations
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from enum import Enum

class EventType(str, Enum):
    MEETING = "meeting"
    TASK = "task"

class CalendarEvent(BaseModel):
    id: str = Field(..., description="Unique event ID (prefixed with type)")
    title: str = Field(..., description="Event title")
    start: datetime = Field(..., description="Start time")
    end: datetime = Field(..., description="End time")
    all_day: bool = Field(False, description="Is all day event")
    type: EventType = Field(..., description="Event type")
    status: Optional[str] = Field(None, description="Status (e.g. todo, done)")
    metadata: dict = Field(default_factory=dict, description="Additional data (original ID, assignee, etc.)")
